external/android-emugl/host/tools/emugen/ApiGen.cpp

```diff
@@ -1161,6 +1205,41 @@ int ApiGen::genDecoderImpl(const std::string &filename)
 
         } // pass;
         fprintf(fp, "\t\t\tSET_LASTCALL(\"%s\");\n", e->name().c_str());
+	if (strcmp(m_basename.c_str(), "gles2") == 0) {
+	    if (strcmp(e->name().c_str(), "glAttachShader") == 0){
+		fprintf(fp, "\n\
+			\t\t\tm_lock.lock();\n\
+			m_shaders.insert({var_shader, 1});\n\
+			m_lock.unlock();\n");
+	    } else if(strcmp(e->name().c_str(), "glDeleteProgram") == 0){
+		fprintf(fp, 
+			"\t\t\tm_lock.lock(); \n"
+			"\t\t\tauto pro = m_programs.find(var_program); \n"
+			"\t\t\tif (pro == m_programs.end()) \n"
+			"\t\t\t{\n"
+			"\t\t\t\tDEBUG(\"This should not happened!\"); \n"
```

Comment:
What should not happen?

Reply:
Sorry for confusing debug info, I will replace it.